src/controller/CHIPDevice.h

```diff
@@ -376,6 +326,32 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
         return LoadSecureSessionParametersIfNeeded(loadedSecureSession);
     };
 
+    InteractionPeer * GetInteractionPeer() { return &mInteractionPeer; }
+
+    //////////// InteractionPeerDelegate Implementation ///////////////
+
+    /**
+     * @brief
+     *   Send the provided message to the device
+     *
+     * @param[in] protocolId  The protocol identifier of the CHIP message to be sent.
+     * @param[in] msgType     The message type of the message to be sent.  Must be a valid message type for protocolId.
+     * @param [in] sendFlags  SendMessageFlags::kExpectResponse or SendMessageFlags::kNone
+     * @param[in] message     The message payload to be sent.
+     *
+     * @return CHIP_ERROR   CHIP_NO_ERROR on success, or corresponding error
+     */
+    CHIP_ERROR SendMessage(Protocols::Id protocolId, uint8_t msgType, Messaging::SendFlags sendFlags,
+                           System::PacketBufferHandle && message) override;
+
+    NodeId GetDeviceId() const override { return mDeviceId; }
+
+    FabricIndex GetFabricIndex() const override { return mFabricIndex; }
+
+    SecureSessionHandle * GetSesssionHandle() override { return &mSecureSession; }
```

Comment:
Typo on `Session` in `GetSessssionHandle`.

This assumes a single session per device (which may be correct, but we are not addressing the lifecycle of that session in the change here)

Reply:
As to the lifecycle of a session, it's supposed to be a simple combination of {fabric-id, node-id, key-id} and be passed by value or const reference. Unfortunately the current interaction model code are accepting pointers and use `nullptr` to create default values. It's tracked in https://github.com/project-chip/connectedhomeip/issues/6792. Fixing this will require another refactor in the `app/` directory and is beyond the scope of this PR.